nrc_ws/src/nrc_nav/src/nrc_drive_dr.py

```diff
@@ -0,0 +1,69 @@
+#! /usr/bin/env python
+
+import rospy
+
+import time
+import csv
+from numpy import genfromtxt
+
+from nrc_msgs.msg import DriveCommand
+
+# Global robot object
+command_pub = None
+# global time
+start_time = time.time()
+# time index for trajectory gen instructions
+instruction_index = 0
+# instructions pulled in at the start of code running
+instructions = None
+
+
+def generate_drive_command(timer_event):
+    # make sure we can access the global time
+    global start_time
+    # get local time at every tick to compare to the path-gen (in seconds)
+    local_time = time.ctime(start_time)
+
+    # use local time to update most recent instruction
+    while instructions[instruction_index + 1][0] < local_time:
+        # if current time has passed the next instruction's time, switch "most recent" to it
+        instruction_index += 1
```

Comment:
You'll want to add some sort of check here to make sure you don't exceed the length of the instructions array